src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/FootnoteServiceTests.cs

```diff
@@ -32,33 +34,742 @@ namespace GovUk.Education.ExploreEducationStatistics.Admin.Tests.Services
+
+            var contextId = Guid.NewGuid().ToString();
+
+            await SeedDatabase(contextId, release, subjects: ListOf(subject), releaseSubjects: ListOf(releaseSubject));
+
+            Either<ActionResult, Footnote> result = await CreateFootnoteWithConfiguration(
+                release.Id,
+                contextId,
+                string.Empty,
+                SetOf<Guid>(),
+                SetOf<Guid>(),
+                SetOf<Guid>(),
+                SetOf<Guid>(),
+                SetOf(subject.Id));
+
+            result.AssertRight();
+            Assert.Single(result.Right.Subjects);
+            Assert.Equal(subject.Id, result.Right.Subjects.Single().SubjectId);
+        }
+
+        [Fact]
+        public async Task CreateFootnote_FilterNotLinkedToRelease_ReturnsFootnoteSpecificationsAreInvalidValidationResult()
+        {
+            Release release = _fixture.DefaultStatsRelease().Generate();
+            Subject subject = _fixture.DefaultSubject().Generate();
+            ReleaseSubject releaseSubject = _fixture.DefaultReleaseSubject().WithRelease(release).WithSubject(subject).Generate();
+
+            var contextId = Guid.NewGuid().ToString();
+
+            await SeedDatabase(contextId, release, subjects: ListOf(subject), releaseSubjects: ListOf(releaseSubject));
+
+            Either<ActionResult, Footnote> result = await CreateFootnoteWithConfiguration(
+                release.Id,
+                contextId,
+                string.Empty,
+                SetOf(Guid.NewGuid()),
+                SetOf<Guid>(),
+                SetOf<Guid>(),
+                SetOf<Guid>(),
+                SetOf<Guid>());
+
+            result.AssertBadRequest(FootnoteSpecificationsAreInvalid);
+        }
+
+        [Fact]
+        public async Task CreateFootnote_FilterIsLinkedToRelease_CreatesFootnoteWithCorrectLinks()
+        {
+            Release release = _fixture.DefaultStatsRelease().Generate();
+            Filter filter = _fixture.DefaultFilter().Generate();
+            Subject subject = _fixture.DefaultSubject().WithFilters(new List<Filter>() { filter }).Generate();
+            ReleaseSubject releaseSubject = _fixture.DefaultReleaseSubject().WithRelease(release).WithSubject(subject).Generate();
+
+            var contextId = Guid.NewGuid().ToString();
+
+            await SeedDatabase(contextId, release, subjects: ListOf(subject), releaseSubjects: ListOf(releaseSubject));
+
+            Either<ActionResult, Footnote> result = await CreateFootnoteWithConfiguration(
+                release.Id,
+                contextId,
+                string.Empty,
+                SetOf(filter.Id),
+                SetOf<Guid>(),
+                SetOf<Guid>(),
+                SetOf<Guid>(),
+                SetOf<Guid>());
+
+            result.AssertRight();
+            Assert.Single(result.Right.Filters);
+            Assert.Equal(filter.Id, result.Right.Filters.Single().FilterId);
+        }
```

Comment:
I don't see there being much benefit of having unit tests for all the 'success' cases at a fine grained level like this. Can't we just trust `CreateFootnote_WithFiltersAndIndicatorsAndSubjectsWhichAreAllLinkedToRelease_CreatesFootnoteWithCorrectLinks` would fail if there was an issue and get rid of all the success permutations for Filter/FilterGroup/Indicator/Subject like this one?

Reply:
I'm not 100% convinced. I did think of this at the time but then decided that technically we wouldn't be covering the cases where we don't supply any other id's other than, say, filterIds etc.... For instance, the underlying logic might be accidentally requiring for all these parameters to be populated with at least 1 ID. I might be doing a `.Single` rather than a `.First` for example, which would cause it to fail if I haven't passed in some IDs. That's a poor way of explaining it - but hopefully it makes sense why I decided it was more robust to test all permutations :) 

I'm happy to remove them if we feel strongly about it - but I'd also say they're there now so there isn't much harm in keeping them? And they do provide some benefit

Reply:
Discussed, going to keep them